### PR TITLE
rename SingleArray -> MaybeArray

### PR DESCRIPTION
**Pull request by burrbull**
cc @Emilgardis 
What you think about this?

### CODE REVIEW

**Review by Emilgardis — APPROVED** [review]
yeah, this is better. Similar to Either

bors r+

File: svd-rs/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 ## Unreleased
 
 - skip serializing `values` in `EnumeratedValues` if empty
+- rename `SingleArray` to `MaybeArray`, add `names` function for arrays
```
Comment (Emilgardis) [inline svd-rs/CHANGELOG.md:11]: you can just rename the entry for SingleArray directly, instead of having two changelog entries for it

**Review by Emilgardis — CHANGES_REQUESTED** [review]

File: svd-rs/CHANGELOG.md (hunk 1 above)
Comment (burrbull) [reply svd-rs/CHANGELOG.md]: Done

**Review by Emilgardis — APPROVED** [review]
bors r+

**Pull request merged**